src/main/java/sx/blah/discord/util/MessageHistoryRange.java

```diff
@@ -0,0 +1,280 @@
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     Discord4J is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Lesser General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Lesser General Public License
+ *     along with Discord4J.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package sx.blah.discord.util;
+
+import sx.blah.discord.handle.impl.obj.Channel;
+import sx.blah.discord.handle.obj.IChannel;
+import sx.blah.discord.handle.obj.IMessage;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This class represents a range of a channel's message history. It includes a start and endpoint to clearly define the
+ * range of messages. These endpoints can be either timestamps or messages, and can be inclusive or exclusive. This
+ * class also contains some logic regarding getting the first message of the range (the message closest to `start` that
+ * is in range).
+ * It is used in {@link MessageHistoryBuilder} and {@link MessageHistoryIterator} to define the range being requested.
+ *
+ * @see MessageHistoryBuilder
+ * @see MessageHistoryIterator
+ * @see Point
+ */
+public class MessageHistoryRange {
+
+	private final Channel channel;
+
+	private final Point start;
+	private final Point end;
+
+	private final boolean chronological;
+
+	public MessageHistoryRange(IChannel channel, Point start, Point end) {
+		this.channel = (Channel) channel;
+
+		this.start = start;
+		this.end = end;
+
+		chronological = start.isBefore(end);
+	}
+
+	/**
+	 * Gets the channel this range is defined in.
+	 *
+	 * @return the channel of this range
+	 */
+	public Channel getChannel() {
+		return channel;
+	}
+
+	/**
+	 * Gets the {@link Point} object that defines this range's starting point
+	 *
+	 * @return the starting point
+	 */
+	public Point getStart() {
+		return start;
+	}
+
+	/**
+	 * Gets the {@link Point} object that defines this range's ending point
+	 *
+	 * @return the ending point
+	 */
+	public Point getEnd() {
+		return end;
+	}
+
+	/**
+	 * Returns true if the starting point is chronologically before the ending point.
+	 * NOTE: A false value from this method corresponds to the way message histories are viewed in Discord, starting at
+	 * the most recent message and ending at the beginning of the channel's history. A true value from this method would
+	 * be the opposite.
+	 *
+	 * @return whether this range is chronological
+	 */
+	public boolean isChronological() {
+		return chronological;
+	}
+
+	/**
+	 * Gets the first message considered "in range" for this range. The first message is the one closest to `start`.
+	 * Equivalent to `#getStart().getMessage()` if a message was supplied as the starting point.
```

Comment:
If anything it should be `{@link #getStart()}.{@link Point#getMessage() getMessage()}`